src/getLoans.ts

```diff
@@ -39,33 +39,44 @@ async function getUserReservesData(user: string, lambdaWallet: Signer) {
 
   const reservesList = await poolDataProviderContract.getAllReservesTokens(); // Get all reserve tokens
   const userReservesData = [];
+  let collateralAsset = null;
 
   for (const reserve of reservesList) {
     const userReserveData = await poolDataProviderContract.getUserReserveData(
       reserve.tokenAddress,
       user
     );
-    const { currentATokenBalance, currentStableDebt, currentVariableDebt } =
-      userReserveData;
-
-    if (
-      currentATokenBalance > BigInt(0) ||
-      currentStableDebt > BigInt(0) ||
-      currentVariableDebt > BigInt(0)
-    ) {
+    const [
+      currentATokenBalance,
+      currentStableDebt,
+      currentVariableDebt,
+      ,
+      ,
+      ,
+      ,
+      ,
+      isCollateral,
+    ] = userReserveData;
+
+    if (isCollateral) {
+      collateralAsset = reserve.tokenAddress;
+    }
+    const hasDebt =
+      currentStableDebt > BigInt(0) || currentVariableDebt > BigInt(0);
+
+    if (hasDebt) {
       userReservesData.push({
+        currentATokenBalance,
         reserveToken: reserve.tokenAddress,
-        collateralAsset:
-          currentATokenBalance > BigInt(0) ? reserve.tokenAddress : null,
-        debtAsset:
-          currentStableDebt > BigInt(0) || currentVariableDebt > BigInt(0)
-            ? reserve.tokenAddress
-            : null,
+        debtAsset: reserve.tokenAddress,
       });
     }
   }
 
-  return userReservesData;
+  return userReservesData.map((item) => ({
+    ...item,
+    collateralAsset,
+  }));
```

Comment:
rel to moving the collateral to the array.
```suggestion
  return userReservesData;
```